### ISSUE TITLE
fink-bld SHELL conflicts with numpy.distutils

### ISSUE THREAD

**Issue by dhomeier** [fink/fink-distributions#121]
Hi,

this problem has come up a couple of times in the past months, but tends to reappear (see links below)...
The issue, briefly, is that numpy's own `distutils` module, and any package using `numpy.distutils`, by default spawns all subprocesses in the shell `$SHELL` (in `numpy/distutils/exec_command.py`):
```
    if os.name == 'posix' and use_shell:
        # On POSIX, subprocess always uses /bin/sh, override
        sh = os.environ.get('SHELL', '/bin/sh')
        if is_sequence(command):
            command = [sh, '-c', ' '.join(command)]
        else:
            command = [sh, '-c', command]
```
Any suggestions how to deal with this in a more sustainable way (and if it's just adding a new `SetSHELL` field for convenience)? It's hard to track down if you are not aware of the problem, as even
```
DISTUTILS_DEBUG=1 %p/bin/python2.7 -d setup.py build
```
only gives you a vague pointer to the source of the failure.
I am considering reporting the issue upstream so numpy will at least test explicitly if `os.environ.get('SHELL')` produces a working command, but they might respond that building with `SHELL=/usr/bin/false` is a really bad idea™...

- [numexpr-py34-2.6.4-1 failed](https://sourceforge.net/p/fink/mailman/message/36280059/)

- [broken python packages](https://sourceforge.net/p/fink/mailman/message/36104712/)

- [numpy test failures](https://sourceforge.net/p/fink/package-submissions/4930/#467b)

**Comment by dmacks** [comment]
It's not just us. FreeBSD has the same problem and patched in in their distro. See [FreeBSD Bug #225156](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=225156)

**Comment by nieder** [comment]
I locally patched my numpy-py35 to include the SHELL fix from FreeBSD and then tried to build numexpr-py35 (all on 10.13/Xcode9.4) against the patched numpy-py. The numexpr-py build succeeds with the patched numpy-py. This patch seems like the correct way to go.
````patch
--- numpy/distutils/exec_command.py.orig	2017-09-29 20:10:10 UTC
+++ numpy/distutils/exec_command.py
@@ -231,7 +231,7 @@ def _exec_command(command, use_shell=Non
 
     if os.name == 'posix' and use_shell:
         # On POSIX, subprocess always uses /bin/sh, override
-        sh = os.environ.get('SHELL', '/bin/sh')
+        sh = '/bin/sh'
         if is_sequence(command):
             command = [sh, '-c', ' '.join(command)]
         else:
````

**Comment by nieder** [comment]
I merged #218. Does that take care of this issue, or is it only a workaround?

**Comment by dhomeier** [comment]
Fixed in #218 and upstream.
